rest/route/github.go

```diff
@@ -352,18 +355,23 @@ func (gh *githubHookApi) tryDequeueCommitQueueItemForPR(pr *github.PullRequest)
 
 	projRef, err := gh.sc.GetProjectWithCommitQueueByOwnerRepoAndBranch(*pr.Base.Repo.Owner.Login, *pr.Base.Repo.Name, *pr.Base.Ref)
 	if err != nil {
-		return errors.Wrapf(err, "can't find projectRef for %s/%s, branch %s", *pr.Base.Repo.Owner.Login, *pr.Base.Repo.Name, *pr.Base.Ref)
+		if adb.ResultsNotFound(err) {
+			return nil
+		}
+		return errors.Wrapf(err, "can't find valid project for %s/%s, branch %s", *pr.Base.Repo.Owner.Login, *pr.Base.Repo.Name, *pr.Base.Ref)
 	}
 
 	_, err = gh.sc.CommitQueueRemoveItem(projRef.Identifier, strconv.Itoa(*pr.Number))
-	// it's okay if there's no matching commit queue or it's not on the queue: there's nothing for us to do
-	if err != nil && !adb.ResultsNotFound(err) {
+	if err != nil && isValidTryDequeueError(err.Error()) {
```

Comment:
I hear your concern about determining the error type. 

Maybe an alternative approach would be to first check if the item is on the queue (through a new method on the commit queue connector) and return if not. This way, when CommitQueueRemoveItem errors we know it's because of a real problem.